osu.Game/Online/API/APIAccess.cs

```diff
@@ -148,7 +148,7 @@ private void run()
                         // The Success callback event is fired on the main thread, so we should wait for that to run before proceeding.
                         // Without this, we will end up circulating this Connecting loop multiple times and queueing up many web requests
                         // before actually going online.
-                        while (State != APIState.Online)
+                        while (State > APIState.Offline)
```

Comment:
This never exits, causing the API to never handle requests.

Reply:
under what circumstances?

Reply:
After the api becomes `Online`. Basically no requests besides the user one (above) happen.

Reply:
oops